Library/Homebrew/download_strategy.rb

```diff
@@ -1169,4 +1166,11 @@ def self.detect_from_symbol(symbol)
       raise "Unknown download strategy #{symbol} was requested."
     end
   end
+
+  def self.require_aws_sdk
+    require "aws-sdk-s3"
+  rescue LoadError
+    Homebrew.install_gem_setup_path! "aws-sdk-s3", "~> 1.8", "ruby"
```

Comment:
Don't pass `ruby` here.

Reply:
Completely leave off the argument? I tried passsing `nil` and it threw an error at me.

Reply:
Try to omit the argument entirely and if that fails post both that and the `nil` errors here and we'll talk about an API fix. Thanks!

Reply:
After uninstalling the gem from the Homebrew bundle...

With `Homebrew.install_gem_setup_path! "aws-sdk-s3", "~> 1.8"` I get:

```
==> Installing or updating 'aws-sdk-s3' gem
Fetching: aws-sdk-s3-1.8.2.gem (100%)
Successfully installed aws-sdk-s3-1.8.2
1 gem installed
Error: The 'aws-sdk-s3' gem is installed but couldn't find 'aws-sdk-s3' in the PATH:
```

With `Homebrew.install_gem_setup_path! "aws-sdk-s3", "~> 1.8", nil` I get:

```
==> Installing or updating 'aws-sdk-s3' gem
Fetching: aws-sdk-s3-1.8.2.gem (100%)
Successfully installed aws-sdk-s3-1.8.2
1 gem installed
Error: no implicit conversion of nil into String
```

Reply:
Any advice, @MikeMcQuaid?